### DISCUSSION TITLE
React native using Custom Register

### DISCUSSION THREAD

**Discussion by abdelrahman-muntaser** [react-hook-form/react-hook-form#7103] [General]
When I am trying to use custom register on react-native TextInput component, I am getting the error "t.split is not a function", but this was working fine for me on version 6.15.5,  when i checked the documentation, I didn't find anything there explaining how to use custom register with the new one,

any Ideas

**Comment by bluebill1049** [comment]
you can do something like below: 

```tsx
const {onChange, onBlur} = register('test')

<TextInput onChange={onChange} onBlur={onBlur} />
```
